modules/datadog/main.tf

```diff
@@ -1,44 +1,92 @@
 # Datadog Operator
-
-locals {
-  datadog_site = "datadoghq.eu"
-  datadog_operator_helm_values = concat(
-    [{ name = "site", value = local.datadog_site }],
-  var.datadog_operator_helm_values)
-}
-module "datadog_operator" {
-  source  = "aws-ia/eks-blueprints-addon/aws"
-  version = ">= 1.0"
-
+resource "helm_release" "datadog_operator" {
   name             = "datadog-operator"
   repository       = "https://helm.datadoghq.com"
   description      = "Open source Kubernetes Operator that enables you to deploy and configure the Datadog Agent in a Kubernetes environment"
   chart            = "datadog-operator"
   namespace        = var.namespace
-  max_history      = 10
-  chart_version    = try(var.datadog.operator_chart_version, "1.8.1") # github-releases/DataDog/datadog-operator
+  max_history      = 3
+  version          = try(var.datadog_operator.chart_version, "2.9.2") # github-releases/DataDog/datadog-operator
   atomic           = true
+  cleanup_on_fail  = true
   create_namespace = true
 
-  set = local.datadog_operator_helm_values
+  values = concat([
+    <<-YAML
+    clusterName: ${var.cluster_name}
+    site: datadoghq.eu
+
+    apiKeyExistingSecret: datadog-keys
+    appKeyExistingSecret: datadog-keys
+
+    datadogAgent:
+      enabled: true
+    datadogDashboard:
+      enabled: true
+    datadogGenericResource:
+      enabled: true
+    datadogMonitor:
+      enabled: true
+    remoteConfiguration:
+      enabled: true
+
+    datadogCRDs:
+      crds:
+        datadogAgents: true
+        datadogMetrics: true
+        datadogPodAutoscalers: true
+        datadogMonitors: true
+        datadogSLOs: false
+        datadogDashboards: true
+        datadogGenericResources: true
+
+    resources:
+      requests:
+        cpu: 50m
+        memory: 128Mi
+
+    watchNamespaces:
+      - ""
+
+    clusterRole:
+      allowReadAllResources: true
+
+    YAML
+  ], try(var.datadog_operator.values, []))
+
+  dynamic "set" {
+    for_each = try(var.datadog_operator.set, [])
+
+    content {
+      name  = set.value.name
+      value = set.value.value
+      type  = try(set.value.type, null)
+    }
+  }
```

Comment:
are we sure that both approaches are capable of override the hardcoded yaml? e.g. if we want to patch the resources to add more cpu

Reply:
Davide mentioned that `dynamic "set"` works, but did we test that concating two lists also does the job?

Reply:
yes, we use it everywhere

Reply:
values is a list of strings and they are processed in order as with helm directly.